src/Surfnet/StepupSelfService/SelfServiceBundle/Service/RemoteVetting/IdentityProviderFactory.php

```diff
@@ -66,4 +78,18 @@ public function create($name)
 
         throw new InvalidRemoteVettingIdentityProviderException(sprintf("Invalid IdP requested '%s'", $name));
     }
+
+
+    /**
+     * @param string $name
+     * @return array
+     */
+    public function getAttributes($name)
+    {
+        if (array_key_exists($name, $this->attributeMapping)) {
+            return $this->attributeMapping[$name];
+        }
+
+        throw new InvalidRemoteVettingIdentityProviderException(sprintf("Invalid IdP requested '%s'", $name));
```

Comment:
Suggestion to make the message more meaningful:

Unable to find the attribute mapping for an unknown IdP identified by '%s'

Reply:
Done